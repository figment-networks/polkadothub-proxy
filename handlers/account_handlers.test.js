const accountHandlers = require('./account_handlers');
const {createApi} = require('../tests/shared');

// These actually DO hit the API
describe('integration', () => {
  let call = {request: {height: 2003930, address: 'DSpbbk6HKKyS78c4KDLSxCetqbwnsemv2iocVXwNe2FAvWC'}};
  let api;

  beforeAll(async (done) => {
    api = await createApi();
    done()
  });

  describe('getIdentity', () => {
    call = {request: {address: 'DSpbbk6HKKyS78c4KDLSxCetqbwnsemv2iocVXwNe2FAvWC'}};

    it('returns a result that matches snapshot', async () => {
      expect(await accountHandlers.getIdentity(api, call)).toMatchSnapshot();
    });
  });

  describe('getByHeight', () => {
    it('returns a result that matches snapshot', async () => {
      expect(await accountHandlers.getByHeight(api, call)).toMatchSnapshot();
    });
  });
});
