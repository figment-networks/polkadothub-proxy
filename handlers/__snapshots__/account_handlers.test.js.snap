// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration getByHeight returns a result that matches snapshot 1`] = `
Object {
  "account": Object {
    "feeFrozen": "0",
    "free": "0",
    "miscFrozen": "0",
    "nonce": 0,
    "referendumCount": 0,
    "reserved": "0",
  },
}
`;

exports[`integration getIdentity returns a result that matches snapshot 1`] = `
Object {
  "identity": Object {
    "deposit": "10000000000000",
    "displayName": "DragonStake üê≤",
    "emailName": "dragonstake@protonmail.com",
    "legalName": "DragonStake",
    "riotName": "@derfredy:matrix.org",
    "twitterName": "
@DragonStake",
    "webName": "https://dragonstake.io",
  },
}
`;
